### PR TITLE
tabbed carousel

### LINKED ISSUE #25: Tabbed carousel block
Similar to Volvo Trucks' [tabbed-carousel block](https://www.volvotrucks.us/trucks/vhd/), this will be a new block for Mack Trucks.

### Requirements
- Full width dark background (graphite colour)
- Section contains headline
- Image with a caption
- Tab controls are placed below the image, with a red indicator for the active
- When the tab controls don’t fit in the viewport, they will be scrollable as in #13* and defined in Volvo Experience System:
> image
> On small screens, if the tabs are wider than screen, the user can side scroll to see all tabs.
- When the tab controls don’t fit in the viewport, they will be scrolled into view when active
- Use scroll-snap ([same as on trucks line-up, on the trucks images](https://github.com/Netcentric/vg-volvotrucks-us-rd/issues/36)), for the tabs

#### Image
- On mobile and tablet, the image is full width
- The image won’t expand the content container (maximum 1040px)
- On mobile and tablet, the caption text will be shown below the image
- On desktop, the image caption text will be shown [above a black/transparent gradient](https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?type=design&node-id=2243-159126&mode=dev) to ensure contrast
- On all platforms, the image caption text will be sized to maximum 572px (55% of the maximum content container)
- On desktop, the image caption text will have a margin-left of 32px and a 16px margin-bottom

### PR DESCRIPTION
**Pull request by kesiah**
Fix #25

Test URLs:
- Before: https://main--vg-macktrucks-com-rd--netcentric.hlx.page/drafts/laura/tabbed-carousel
- After: https://25-tabbed-carousel--vg-macktrucks-com-rd--netcentric.hlx.page/drafts/laura/tabbed-carousel

### CODE REVIEW

File: blocks/v2-tabbed-carousel/v2-tabbed-carousel.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        const { x, width } = e.currentTarget.getBoundingClientRect();
+        Object.assign(navigationLine.style, {
+          left: `${x + tabNavigation.scrollLeft - navigationX}px`,
+          width: `${width}px`,
+        });
+      });
+
+      button.addEventListener('mouseout', () => {
+        timeout = setTimeout(() => {
+          const { x: navigationX } = tabNavigation.getBoundingClientRect();
```
Comment (TomaszDziezykNetcentric) [inline blocks/v2-tabbed-carousel/v2-tabbed-carousel.js:27]: We have a very similar code in the `mouseover` handler. Can we move it to a separate function like:
```
const navigatioLineRestyle = (item, item2) => { // have no idea how to call the arguments better
        const { x: navigationX } = item2.getBoundingClientRect();
        const { x, width } = item.getBoundingClientRect();
        Object.assign(navigationLine.style, {
          left: `${x + tabNavigation.scrollLeft - navigationX}px`,
          width: `${width}px`,
        });
}

button.addEventListener('mouseover', (e) => {
        clearTimeout(timeout);
        navigatioLineRestyle(e.target, tabNavigation);
}

button.addEventListener('mouseout', (e) => {
    timeout = setTimout(() => {
              const activeItem = document.querySelector(`.${blockName}__navigation-item.active`);
        navigatioLineRestyle(activeItem, tabNavigation)
    });
}
```

File: blocks/v2-tabbed-carousel/v2-tabbed-carousel.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  const carouselItems = document.querySelector(`.${blockName}__items`);
+  const navigation = document.querySelector(`.${blockName}__navigation`);
+  const navigationLine = document.querySelector(`.${blockName}__navigation-line`);
+
+  [carouselItems, navigation].forEach((c) => c.querySelectorAll('.active').forEach((i) => i.classList.remove('active')));
+  carouselItems.children[index].classList.add('active');
+  navigation.children[index].classList.add('active');
+
+  if (navigationLine) {
+    const { x: navigationX } = navigation.getBoundingClientRect();
```
Comment (TomaszDziezykNetcentric) [inline blocks/v2-tabbed-carousel/v2-tabbed-carousel.js:58]: You reuse the function from the prev comment:
```
navigatioLineRestyle(activeNavigationItem, navigation)
```

Comment (TomaszDziezykNetcentric) [inline blocks/v2-tabbed-carousel/v2-tabbed-carousel.js:58]: You reuse the function from the prev comment:
```
navigatioLineRestyle(activeNavigationItem, navigation)
```

**Review by TomaszDziezykNetcentric — CHANGES_REQUESTED** [review]

File: blocks/v2-tabbed-carousel/v2-tabbed-carousel.css [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  --navigation-line-active-color: var(--c-accent-red);
+  --navigation-padding: 24px 16px;
+  --navigation-line-top: 21px;
+  --navigation-line-height: 3px;
+
+  background-color: var(--bg-color);
+}
+
+.redesign-v2 .section.v2-tabbed-carousel-container {
+  padding: 0;
```
[the remark this answers is not in the document]
Comment (kesiah) [reply blocks/v2-tabbed-carousel/v2-tabbed-carousel.css]: Removed the styles, but then the spacings are not looking good when we add a text over the tabbed carousel. But that's not an issue of this block, so out of scope ;P

**Review by markovukiceviccn — APPROVED** [review]

**Pull request merged**